### ISSUE TITLE
SDK 37 library upgrades

### ISSUE THREAD

**Issue by esamelson** [expo/expo#7006]
Creating this issue to track third-party library upgrades for SDK 37. We'll most likely want to make the following upgrades:

- [x] "@react-native-community/netinfo": "5.3.3",
- [x] "@react-native-community/datetimepicker": "2.2.1",
- [x] "@react-native-community/masked-view": "0.1.6",
- [x] "@react-native-community/viewpager": "3.3.0",
- [x] "react-native-gesture-handler": "~1.5.6",
- [x] "react-native-reanimated": "~1.7.0",
- [x] "react-native-screens": "2.2.0",
- [x] "react-native-svg": "11.0.1",
- [x] "react-native-view-shot": "3.0.2",
- [x] "react-native-webview": "8.0.6",
- [x] "react-native-appearance": "~0.3.2",
- [x] "react-native-safe-area-context": "0.7.2",
- [x] "react-native-shared-element": "~0.5.6",

As of this writing, we have the most up-to-date version of react-native-maps already, and I believe we're blocked on upgrading lottie-react-native by Swift. Let's wait until the week of Feb 17 to do these so that we can capture any newer updates that happen between now and then.

**Comment by jaulz** [comment]
@esamelson could we use `@react-native-community/netinfo@~5.5.0`? Web supports has recently [landed](https://github.com/react-native-community/react-native-netinfo/releases) 😊

**Comment by tsapeta** [comment]
@jaulz Yes, we're going to update all those libs to the newest possible versions. Sometimes we're blocked because of React Native version or just like `lottie-react-native` that we can't upgrade because it uses Swift, but in case of NetInfo it looks like it's safe to update to `5.5.0` so don't worry 🙂

**Comment by jaulz** [comment]
Great, thanks for the prompt feedback! :blush:

**Comment by todorone** [comment]
@tsapeta @esamelson  Btw, new `lottie-react-native` has significant performance regression on iOS, so maybe leave old version for now which is more performant?
https://github.com/airbnb/lottie-ios/issues/895

**Comment by tsapeta** [comment]
@todorone Yeah, thanks for letting us know. Anyway, as I said in my previous comment: we can't upgrade `lottie-react-native` because the new version uses Swift that we can't use in our repo yet and don't plan to support this in the near future, so we need to stick with `lottie-ios@2.5.0` which seems to be fine in terms of the performance.

**Comment by Hirbod** [comment]
Please, if possible, pick the most up to date version of react native screens. They made huge process and beta 10 is a big milestone with many many crashes and fixes. You guys pinned beta.2 in the ticket

**Comment by tsapeta** [comment]
@Hirbod Yes, it will be upgraded to the most recent one, my colleagues at @software-mansion are still working and testing another version that is not out yet but we're close to publishing `beta.11` of `react-native-screens` 🙂

**Comment by Hirbod** [comment]
I love u guys, thanks!!!

**Comment by alaminopu** [comment]
Please if possible also upgrade `react-native-branch`. Branch integration is currently broken. It's not working as expected.

**Comment by brentvatne** [comment]
@alaminopu - can you please point us to an issue that explains what problems you have encountered with branch?

**Comment by alaminopu** [comment]
@brentvatne Thanks for your reply. I haven't been able to make it work with the latest version. I opened an issue https://github.com/expo/expo/issues/6815 before.

**Comment by ken0x0a** [comment]
@esamelson Is it possible to use "react-native-gesture-handler" ~1.6.0 as it works good on "WEB".

**Comment by brentvatne** [comment]
@ken0x0a - yes this is already linked to from this issue https://github.com/expo/expo/pull/7121

**Comment by LinusU** [comment]
Could we also update react-native-web from 0.7.x to 0.8.x? Thanks ❤️ 

Can the community help out with the upgrades? I'd be happy to submit a PR ☺️

**Comment by brentvatne** [comment]
@LinusU 

> Could we also update react-native-web from 0.7.x to 0.8.x? Thanks ❤️ 

maybe I'm misunderstanding this - it looks like we are using ~0.11.7 for react-native-web. Maybe a typo and you meant another package?

> Can the community help out with the upgrades? I'd be happy to submit a PR ☺️

Which ones did you have in mind? Typically we just run a script to pull in the latest version of a particular package: https://github.com/expo/expo/blob/2f13ef01f8ab520dc4b7a08e9e4a03e27bc77431/tools/expotools/src/commands/UpdateVendoredModule.ts

It's pretty easy for us to do this and we would do it more often but we need to version them at the same time as each SDK and there is a lot of work required for every SDK release.

**Comment by ken0x0a** [comment]
I guess it would be "~0.11.7" to "^0.12.2".
If some packages extracted from react-native core, like "@react-native-community/masked-view", "AsyncStorage" & ..., is also extracted from expo/react-native, just bumping version works fine I guess.

**Comment by LinusU** [comment]
> maybe I'm misunderstanding this - it looks like we are using ~0.11.7 for react-native-web. Maybe a typo and you meant another package?

Whoops, my bad, this is what happens when I work too much 😁 

> I guess it would be "~0.11.7" to "^0.12.2".

This is it ☝️ 

> It's pretty easy for us to do this and we would do it more often but we need to version them at the same time as each SDK and there is a lot of work required for every SDK release.

Should I interpret this as "it's just a script so it's easier for us to do it when we cut a new SDK release" or "we're busy with a ton of stuff so PR's for that are very welcome"? The one I had in mind now was react-native-web.

I've been re-building or app from scratch with Expo, and specifically Expo Web, to see if it's viable for us to switch to Expo and build our web application from the same code-base. My impressions so far is that there is still a lot happening in the ecosystem and it's a bit early. So I'll likely look into building the SDK myself, so I don't have to wait for the longer normal release cycle, next ☺️ 

There seems to be a ton of bug fixes happening in react-native-screens as well which is already at 2.2.0 whilst we upgraded to 2.0.0 here very recently 💪

**Comment by Hirbod** [comment]
@LinusU  They have just upgraded to 2.1.0 on react-native-screens (its already merged into SDK 37), but new versions of react-native-screens are popping up every second :D - like 2.2.0 now.

I dare not ask for another update, but since every SDK Update will take ~3 months, mostly bug-free versions are pretty important.

**Comment by LinusU** [comment]
@Hirbod yeah, saw that after I commented, and also submitted a PR to bring it up to 2.2.0 ☺️ #7215

**Comment by Hirbod** [comment]
Thats good @LinusU, since 2.1.0 is crashing on android native stack.

**Comment by brentvatne** [comment]
> Should I interpret this as "it's just a script so it's easier for us to do it when we cut a new SDK release" or "we're busy with a ton of stuff so PR's for that are very welcome"? The one I had in mind now was react-native-web.

the hard part isn't really updating the code but testing it and deploying it, so the first interpretation makes sense.

i'm not really sure about what goes in to updating the react-native-web version, this is an @EvanBacon question.

**Comment by congnt171** [comment]
Can you update react-native-webview 9.0.1. It have big update for inject on android. 
Thanks @tsapeta

**Comment by Hirbod** [comment]
@congnt171 I think that train is gone for SDK 37 looking into the Release 37 Todos

**Comment by brentvatne** [comment]
@congnt171 - we are updating to 8.1.2, there are no changes to inject on android from that version to 9.0.0 (just released 20 hours ago) so i imagine this will solve your issue

**Comment by dotjsNet** [comment]
When will you be releasing the expo app with sdk 37.0.0 support, i upgraded and its no longer working because app doesn't have support for this version at the moment.

**Comment by brentvatne** [comment]
you shouldn't upgrade to new sdk verisons until we publish the release notes on blog.expo.io

**Comment by moxorama** [comment]
Is it possible to upgrade react-native-maps to 0.27.1?

**Comment by brentvatne** [comment]
that will have to come in sdk 38, we have already done the upgrades for sdk 37 and will be shipping it soon

**Comment by dotjsNet** [comment]
@brentvatne right but running expo upgrade update ultimately to sdk 37 which could be possible bug with expo upgrade then.

**Comment by brentvatne** [comment]
@dotjsNet - i don't think it does, it updates to sdk 36. i just tried it :O

**Comment by kopax** [comment]
How can I upgrade from expo sdk 36 to 37, before upgrading to expo sdk 38? Is that even possible?

**Comment by tsapeta** [comment]
@kopax Yes, you can find steps to upgrade to 37 at the end of this blog post: https://blog.expo.io/expo-sdk-37-is-now-available-dd5770f066a6

**Comment by MorenoMdz** [comment]
> @todorone Yeah, thanks for letting us know. Anyway, as I said in my previous comment: we can't upgrade `lottie-react-native` because the new version uses Swift that we can't use in our repo yet and don't plan to support this in the near future, so we need to stick with `lottie-ios@2.5.0` which seems to be fine in terms of the performance.

About this, some users report that downgrading the version of lottie animations fixes the compat issue with lottie 2.5 and android, is there a recommended way to do this downgrade?
